pdns/rcpgenerator.cc

```diff
@@ -409,22 +453,33 @@ void RecordTextReader::xfrSvcParamKeyVals(set<SvcParam>& val)
     }
     case SvcParam::port: {
       uint16_t port;
-      xfr16BitInt(port);
+      if (generic) {
+        string v;
+        xfrRFC1035CharString(v);
+        port = (v.at(0) << 8);
```

Comment:
Might be good to check the length so we can have a better error message if it's != 2.